### PR TITLE
Fix recipe for py-onnx-runtime

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-onnx-runtime/package.py [hunk 1]
```diff
@@ -126,4 +127,9 @@ def cmake_args(self):
 
     @run_after("install")
     def install_python(self):
-        PythonPackage.install(self, self.spec, self.prefix)
+        platlib = self.spec["python"].package.platlib
+        with working_dir(self.build_directory):
+            py = which("python")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-onnx-runtime/package.py]: This won't work for `python~pythoncmd` where only the `python3` executable exists. Spack already has a `python` function injected into the module, so you can use `python(...)` without using `which(...)` first.

File: var/spack/repos/builtin/packages/py-onnx-runtime/package.py [hunk 2]
```diff
@@ -126,4 +127,9 @@ def cmake_args(self):
...
+            py("setup.py", "build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-onnx-runtime/package.py]: Direct invocation of `setup.py` is deprecated, see https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html and #27798. We should instead use a build tool like pip.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-onnx-runtime/package.py [hunk 3]
```diff
@@ -126,4 +132,11 @@ def cmake_args(self):
...
     def install_python(self):
-        PythonPackage.install(self, self.spec, self.prefix)
+        """Install everything from build directory."""
+        # Truncated version of PythonPipBuilder.install
+
+        args = PythonPipBuilder.std_args(self) + ["--prefix=" + prefix, "."]
+
+        pip = inspect.getmodule(self).pip
+        with fs.working_dir(self.build_directory):
+            pip(*args)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-onnx-runtime/package.py]:
```suggestion
        args = std_pip_args + ["--prefix=" + prefix, "."]
        with working_dir(self.build_directory):
            pip(*args)
```
This should be sufficient. This is also what we do in `py-tensorflow` and many other packages like this.

**Comment by iarspider** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-onnx-runtime/package.py [hunk 4]
```diff
@@ -3,6 +3,8 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import inspect
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-onnx-runtime/package.py]: You can remove this import

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**